lms/djangoapps/teams/search_indexes.py

```diff
@@ -78,7 +82,11 @@ def engine(cls):
         Return course team search engine (if feature is enabled).
         """
         if cls.search_is_enabled():
-            return SearchEngine.get_search_engine(index=cls.INDEX_NAME)
+            try:
+                return SearchEngine.get_search_engine(index=cls.INDEX_NAME)
+            except ConnectionError as err:
+                logging.error("Error connecting to elasticsearch: {err}".format(err=err))
+                raise ElasticSearchConnectionError
```

Comment:
I'd like to see if we can preserve the traceback of the `ConnectionError` so that callers can see the real source of the error. I *think* this should be as simple as grabbing `sys.exc_info` and raising the old traceback with a new type and value parameter, but I'm not sure. @nedbat seems to be our resident Python expert; maybe he can weigh in?